src/js/gi/selectors/calculator.js

```diff
@@ -8,16 +10,896 @@ const getEligibilityDetails = (state) => {
+    const isPublic = institution.type === 'public';
+    const isPrivate = institution.type === 'private';
+    const isForeign = institution.type === 'foreign';
+
+    // VRE and post-9/11 eligibility
+    const vre911Eligible = (giBillChapter === 31 && eligForPostGiBill === 'yes');
+
+    // VRE-without-post-911 eligibility
+    const onlyVRE = (giBillChapter === 31 && eligForPostGiBill === 'no');
+
+    // Determines benefits tier
+    const tier = (vre911Eligible || serviceDischarge) ? 1 : +cumulativeService;
+
+    const oldGiBill = (
+      giBillChapter === 30
+      || giBillChapter === 1607
+      || giBillChapter === 1606
+      || giBillChapter === 35
+    );
+
+    // Determines whether monthly benefit can only be spent on tuition/fees
+    const activeDutyThirtyOr1607 = (
+      militaryStatus === 'active duty' &&
+      (giBillChapter === 30 || giBillChapter === 1607)
+    );
+    const correspondenceOrFlightUnderOldGiBill =
+      isFlightOrCorrespondence && oldGiBill;
+    const ropOldAndChapter = (
+      ['less than half', 'quarter'].includes(inputs.enrolledOld) &&
+      [30, 35, 1607].includes(giBillChapter)
+    );
+    const onlyTuitionFees =
+      activeDutyThirtyOr1607 ||
+      correspondenceOrFlightUnderOldGiBill ||
+      ropOldAndChapter;
+
+    // The monthly benefit rate for non-chapter 33 benefits
+    switch (giBillChapter) {
+      case 30:
+        if (enlistmentService === '3' || enlistmentService === '2') {
+          monthlyRate = isOJT ? constant.MGIB3YRRATE * 0.75 : constant.MGIB3YRRATE;
+        }
+        break;
+      case 1607:
+        monthlyRate = constant.MGIB3YRRATE * consecutiveService;
+        monthlyRate = isOJT ? monthlyRate * 0.75 : monthlyRate;
+        break;
+      case 1606:
+        monthlyRate = isOJT ? constant.SRRATE * 0.75 : constant.SRRATE;
+        break;
+      case 35:
+        if (isOJT) {
+          monthlyRate = constant.DEARATEOJT;
+        } else if (isFlight) {
+          monthlyRate = 0;
+        } else {
+          monthlyRate = constant.DEARATE;
+        }
+        break;
+      case 31:
+        if (numberOfDependents <= 2) {
+          monthlyRate =
+            constant[`VRE${numberOfDependents}DEPRATE${isOJT ? 'OJT' : ''}`];
+        } else {
+          monthlyRate =
+            constant[`VRE2DEPRATE${isOJT ? 'OJT' : ''}`] +
+            ((numberOfDependents - 2) *
+            constant[`VREINCRATE${isOJT ? 'OJT' : ''}`]);
+        }
+        break;
+      default:
+        monthlyRate = 0;
+    }
+
+    // Calculate the total number of academic terms - getNumberOfTerms
+    if (isOJT || inputs.calendar === 'quarters') {
+      numberOfTerms = 3;
+    } else if (inputs.calendar === 'semesters') {
+      numberOfTerms = 2;
+    } else if (inputs.calendar === 'nontraditional') {
+      numberOfTerms = +inputs.numberNontradTerms;
+    }
+
+    // Set the net price (Payer of Last Resort) - getTuitionNetPrice
+    const tuitionNetPrice = Math.max(0, Math.min(
```

Comment:
The math-min here is to avoid having the tuition netprice go below a certain threshold... it looks like this logic is missing the other part of the min which would the tuition costs.